web/src/app/schedules/ScheduleShiftList.js

```diff
@@ -152,9 +161,14 @@ export default class ScheduleShiftList extends React.PureComponent {
     shifts: [],
   }
 
-  state = {
-    create: null,
-    specifyDuration: false,
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      create: null,
+      specifyDuration: false,
+      startDate: DateTime.fromISO(props.start, { zone: props.zone }) || null,
```

Comment:
Why are we duplicating `start` into state?

Reply:
Since the date can be invalid while they are typing, we don't want to store an invalid timestamp into the URL. It will only update the URL with an ISO timestamp once a valid date is entered into the form field, thus, an invalid date can be stored in state, but only valid dates will be in the URL.

Reply:
Would it be possible to store this in a wrapper component instead? Said component would only trigger `onChange` when it changes to a new _valid_ value, instead of storing mixed state everywhere else.

Reply:
Do we need to do that though? The state is isolated to this component, and will update the URL as soon as it gets a valid date